# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1263954606 -3600
#      Wed Jan 20 03:30:06 2010 +0100
# Node ID cf6a01e0e93fc94bf70d03294044b335f4816d5c
# Parent  ee18258bc00291bd7268a477e436d788fb2eb736
Fix unrequested markers for patch objects

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,14 @@
 2010-01-20  David Bateman  <dbateman@free.fr>
 
+	* plot/__go_draw_axes__.m: Correct an error in drawing of patch
+	objects where unrequested markers were drawn.
+
+
 	* plot/__go_draw_figure__.m: Also plot white foreground and
 	background colors.
 	* plot/plotyy.m: Set axes colors to "none".
 
 	* plot/__go_draw_axes__.m (next_marker):  Remove and all uses of
 	this function.
 	Partially respect the markeredgecolor and markerfacecolor properties.
 	* plot/private/__color__str_rgb__.m: New function
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -919,20 +919,21 @@ function __go_draw_axes__ (h, plot_strea
 	         tmpwith{sidx} = sprintf ("with %s %s %s %s %s %s",
 					  style, lw, pt2, lt, ps, 
 					  colorspec);
                endif
              endif
 
              if (isfield (obj, "markeredgecolor") 
 	         && !strncmp (obj.markeredgecolor, "none", 4))
-               if (facesame && (strncmp (obj.markeredgecolor, "auto", 4)
-		                || ! isnumeric (obj.markeredgecolor) 
-		                || (isnumeric (obj.markeredgecolor) 
-			            && isequal (color, obj.markeredgecolor))))
+               if (facesame && !isempty (pt) 
+                   && (strncmp (obj.markeredgecolor, "auto", 4)
+		       || ! isnumeric (obj.markeredgecolor) 
+		       || (isnumeric (obj.markeredgecolor) 
+			   && isequal (color, obj.markeredgecolor))))
 	         if (sidx == 1 && ((length (style) == 5 
 	                  && strncmp (style, "lines", 5)) 
                          || isempty (style)))
 	           style = strcat (style, "points");
 	           if (isfield (obj, "markersize"))
 	             if (length (mdat) == nc)
 		       m = mdat(i);
 	             else
@@ -943,46 +944,49 @@ function __go_draw_axes__ (h, plot_strea
                      ps = "";
 	           endif
 	           tmpwith{sidx} = sprintf ("with %s %s %s %s %s %s",
 					    style, lw, pt, lt, ps, 
 					    colorspec);
 	         endif
                else
 	         if (!isempty (style))	
-                   if (isempty (tmpwith{sidx}))
+                   if (length(tmpwith) < sidx || isempty (tmpwith{sidx}))
 	             tmpwith{sidx} = sprintf ("with %s %s %s %s",
 					      style, lw, lt, 
 					      colorspec);
                    endif
 	           sidx ++;
 	         endif
-	         if (! mono)
-	           if (strncmp (obj.markeredgecolor, "auto", 4))
-	             colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
-				          round (255*color));
-	           elseif (isnumeric (obj.markeredgecolor) && ! mono)
-	             colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
-				          round (255*obj.markeredgecolor));
+
+                 if (!isempty (pt)) 
+	           if (! mono)
+	             if (strncmp (obj.markeredgecolor, "auto", 4))
+	               colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
+				            round (255*color));
+	             elseif (isnumeric (obj.markeredgecolor) && ! mono)
+	               colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
+				            round (255*obj.markeredgecolor));
+	             endif
 	           endif
-	         endif
-	         style = "points";
-	         if (isfield (obj, "markersize"))
-	           if (length (mdat) == nc)
-		     m = mdat(i);
-	           else
-		     m = mdat;
+	           style = "points";
+	           if (isfield (obj, "markersize"))
+	             if (length (mdat) == nc)
+		       m = mdat(i);
+	             else
+		       m = mdat;
+	             endif
+		     ps = sprintf("pointsize %f", m / 3);
+                   else
+                     ps = "";
 	           endif
-		   ps = sprintf("pointsize %f", m / 3);
-                 else
-                   ps = "";
-	         endif
-	         tmpwith{sidx} = sprintf ("with %s %s %s %s %s %s",
-					  style, lw, pt, lt, ps, 
-					  colorspec);
+	           tmpwith{sidx} = sprintf ("with %s %s %s %s %s %s",
+					    style, lw, pt, lt, ps, 
+					    colorspec);
+                 endif
                endif
              endif
 
              if (isempty (tmpwith))
                withclause{data_idx} = sprintf ("with %s %s %s %s %s",
                                                style, lw, pt, lt, 
                                                colorspec);
              else
